### PR TITLE
feat: support docker language

### PR DESCRIPTION
**Pull request by bxb100**
close: #49

## TODO

- [x] add test suits
- [x] test inner container
	- disable MacOS test, because it's without docker[^1]
	- disable Windows test, because it's without linux kernel[^2]
- [x] test windows

[^1]: https://github.com/actions/runner-images/issues/17
[^2]: https://github.com/orgs/community/discussions/21317

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -24,9 +24,11 @@ home = "0.5.9"
 indicatif = "0.17.8"
 indoc = "2.0.5"
 itertools = "0.13.0"
+md5 = "0.7.0"
 owo-colors = "4.1.0"
 rand = "0.8.5"
 rayon = "1.10.0"
+regex-lite = "0.1.6"
```
Comment (j178) [inline Cargo.toml]: Can we reuse the fancy-regex crate?

Comment (bxb100) [reply Cargo.toml]: Ok, I will fix it in the next commit

**Comment by j178** [comment]
Thanks for your fantastic work! I'll review it soon!

**Comment by bxb100** [comment]
@j178 I want to rewrite all commands to this crate https://github.com/fussybeaver/bollard , so maybe late review /= =/

**Comment by j178** [comment]
@bxb100 I would prefer using the Docker CLI because it is lightweight and doesn't require many dependencies for basic Docker operations, unlike `bollard`, which seems a bit heavy to me. What do you think?

**Comment by bxb100** [comment]
@j178 Ok

**Comment by j178** [comment]
Thanks for your fantastic work here! I'm planning to make some adjustments to the coding style, hope you don't mind. Also, I'm considering simplifying the Docker test by removing the inner Docker test. It seems unnecessary to add a complex test job just for that.

**Comment by j178** [comment]
The `typos-docker` image is around 500MB, it might not be suitable for a test case (It's too slow to run on my local machine).

**Comment by bxb100** [comment]
I considered this and added a `nextest` profile inner `.config/nextest.toml` to prevent test docker(that does not work with `cargo test`).

maybe we can preserve it and add `#[ignored]`, only active this on github CI

**Comment by bxb100** [comment]
I plan to modify `ci-docker.yml` to `ci-expensive.yml`, leverage ignore cfg and nextest filter to disable local test, but work on CI

**Comment by j178** [comment]
(I'll handle the conflicts with master)

**Comment by j178** [comment]
Thanks for your patience, let's merge!

**Pull request merged**